Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/DataSource/Data/AbstractData.cs

```diff
@@ -0,0 +1,59 @@
+using Anvil.CSharp.Core;
+using Anvil.CSharp.Logging;
+using Anvil.Unity.DOTS.Jobs;
+using System.Runtime.CompilerServices;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    //TODO: #136 - Too many responsibilities
+    internal abstract class AbstractData : AbstractAnvilBase
+    {
+        private readonly AccessController m_AccessController;
+        
+        public uint ID { get; }
+        public CancelBehaviour CancelBehaviour { get; }
+        
+        public ITaskSetOwner TaskSetOwner { get; }
+        
+        public AbstractData PendingCancelActiveData { get; }
```

Comment:
:bike: Why not just readonly fields?

Reply:
🤷 I don't know, why readonly fields? 

I've typically been doing readonly getters for anything that isn't private for a while now.

Reply:
Huh, ok you made me look it up and this is a case where the compiler is smart enough.

It doesn't matter they get inlined.
https://stackoverflow.com/a/9843094/640196
(and the answer below that one where they link to an MSDN doc)